docs/classes/request.md

```diff
@@ -2,86 +2,128 @@
 
-Currently it will return one of the three supported versions: `"1.0"`, `"1.1"` or `"2.0"`.
+Returns the HTTP protocol version as a string (e.g. "1.0", "1.1", "2.0").
 
-## `getHeader(string): string | null`
+## `setProtocolVersion(string $protocol)`
 
-Gets the first value of all the headers with that name.
+Sets a new protocol version number for the request.
 
-## `getHeaderArray(string): array<string>`
+## `setHeader(string $name, string | string[] $value)`
 
-Gets an array with headers. HTTP allows for multiple headers with the same name, so this returns an array. Usually only a single header is needed and expected, in this case there is `getHeader()`.
+Sets the named header to the given value.
 
-## `getAllHeaders(): array<array<string>>`
+If header `$name` is not valid or `$value` is invalid, [`\Error`](http://php.net/manual/en/class.error.php) is thrown. 
 
-Returns all the headers in an associative map with the keys being normalized header names in lowercase.
+{:.note}
+> To verify header `$name` and `$value` [`Message::setHeader`](message.md#setheaderstring-name-string--string-value) is used 
 
-## `getParam(string): string | null`
+## `addHeader(string $name, string | string[] $value)`
 
-Gets the first value of all the query string parameters with that name.
+Adds the value to the named header, or creates the header with the given value if it did not exist.
 
-## `getParamArray(string): array<string>`
+If header `$name` is not valid or `$value` is invalid, [`\Error`](http://php.net/manual/en/class.error.php) is thrown. 
 
-Gets an array with the values of the query string parameters with that name.
+{:.note}
+> To verify header `$name` and `$value` [`Message::addHeader`](message.md#addheaderstring-name-string--string-value) is used
 
-## `getAllParams(): array<array<string>>`
+## `removeHeader(string $name)`
 
-Gets the decoded query string as associative array.
+Removes the given header if it exists.
 
-## `getBody(int): Amp\ByteStream\Message`
+## `getBody(): RequestBody`
 
-Returns a representation of the request body. The [`Amp\ByteStream\Message`](http://amphp.org/byte-stream/message) can be `yield`ed to get the actual string.
+Returns the request body. The [`RequestBody`](requestbody.md) allows streamed and buffered access to an [`InputStream`](https://amphp.org/byte-stream/).
 
-The parameter `$bodySize` defaults to `-1` to take the globally configured maximum, otherwise it's the maximum body size.
+## `setBody(RequestBody | InputStream | string | null $stringOrStream)`
 
-There also exists a [`parseBody()`](parsedbody.md) function for processing of a typical HTTP form data.
+Sets the stream for the message body
 
-## `getCookie(string): string | null`
+{:.note}
+> Using a string will automatically set the `Content-Length` header to the length of the given string. Using an [`InputStream`](https://amphp.org/byte-stream/) or [`Body`](requestbody.md) instance will remove the `Content-Length` header.
```

Comment:
The `content-length` header isn't changed in case a stream is passed.

Reply:
To be really honest, I copy-pasted this part from phpdoc of the `setBody` method. 

Should I change the comment or remove `content-length` header for cases when a stream passed to the func?

Reply:
I'm not entirely sure. @trowski Thoughts?

Reply:
The docblock is wrong, it must have not been updated when that behavior was changed. `content-length` is only changed if a string or null is passed. Please change both.